src/test/java/com/spotify/futures/CompletableFuturesTest.java

```diff
@@ -408,6 +408,15 @@ public void combine4_exceptional() throws Exception {
     getCompleted(future);
   }
 
+  @Test(expected=IllegalStateException.class)
+  public void combine4_incomplete() throws Exception {
+    final CompletionStage<String> future = combine(
+        completedFuture("a"), completedFuture("b"), completedFuture("c"),
+        incompleteFuture(),
+            (a, b, c, d)-> a + b + c + d);
+    getCompleted(future);
```

Comment:
nitpicking: Would be nice to use the `exception` rule instead of the annotation so we ensure it throws when we try to `getComplete` it and not before, e.g.
```java
    exception.expectCause(isA(IllegalStateException.class));
    getCompleted(future);
```